spec/classes/puppet_agent_windows_install_spec.rb

```diff
@@ -33,6 +33,28 @@
               is_expected.to contain_file('C:\tmp\install_puppet.bat').with_content(/PID eq 42/)
             }
           end
+          describe 'C:/tmp/puppet-agent-x64.msi' do
+            let(:params) { {
+              :source => 'C:/tmp/puppet-agent-x64.msi',
+            } }
+            it {
+              is_expected.to contain_file('C:\tmp\install_puppet.bat').with_content(
+                               /msiexec.exe \/qn \/norestart \/i "C:\\tmp\\puppet-agent-x64\.msi"/)
+              is_expected.to contain_file('C:\tmp\install_puppet.bat').with_content(/\/l\*v "C:\\tmp\\puppet-\d+_\d+_\d+-\d+_\d+-installer.log"/)
+              is_expected.to contain_file('C:\tmp\install_puppet.bat').with_content(/PID eq 42/)
+            }
+          end
+          describe 'C:\Temp/ Folder\puppet-agent-x64.msi' do
+            let(:params) { {
+              :source => 'C:\Temp/ Folder\puppet-agent-x64.msi',
+            } }
+            it {
+              is_expected.to contain_file('C:\tmp\install_puppet.bat').with_content(
+                               /msiexec.exe \/qn \/norestart \/i "C:\\Temp Folder\\puppet-agent-x64\.msi"/)
+              is_expected.to contain_file('C:\tmp\install_puppet.bat').with_content(/\/l\*v "C:\\tmp\\puppet-\d+_\d+_\d+-\d+_\d+-installer.log"/)
+              is_expected.to contain_file('C:\tmp\install_puppet.bat').with_content(/PID eq 42/)
+            }
+          end
```

Comment:
It failed validation, but passed tests... didn't accept '\/' as valid escape sequence... moved into a simple function.